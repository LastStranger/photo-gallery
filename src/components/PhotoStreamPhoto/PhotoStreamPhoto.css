@import '../../common_styles/colors.css';
@import '../../common_styles/breakpoints.css';

.stream-photo {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 48%;
  margin-bottom: 10px;

  @media (--breakpoint-m) {
    width: 32%;
  }

  @media (--breakpoint-l) {
    width: 24%;
  }
}

.stream-photo-img {
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  width: 100%;
  height: 0;
  padding-bottom: 73.214285714%; /* aspect ratio of mock*/
  background-color: gray(90%);
  border: 1px solid gray(85%);
}
