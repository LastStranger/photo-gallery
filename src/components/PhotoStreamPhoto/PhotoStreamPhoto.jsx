// @flow

import { h } from 'preact';
import NavigationLink from '../NavigationLink';
import './PhotoStreamPhoto.css';

const PhotoStreamPhoto = (props: { id: string, url: string, alt: string, startLoading: Function }) =>
  <NavigationLink onStart={props.startLoading} to={`/${props.id}`} className="stream-photo">
    <div
      aria-label={props.alt}
      role="img"
      className="stream-photo-img"
      style={{ backgroundImage: `url(${props.url})` }}
    />
  </NavigationLink>;

export default PhotoStreamPhoto;
