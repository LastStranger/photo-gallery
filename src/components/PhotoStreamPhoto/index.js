// @flow
import PhotoStreamPhoto from './PhotoStreamPhoto';

export default PhotoStreamPhoto;
