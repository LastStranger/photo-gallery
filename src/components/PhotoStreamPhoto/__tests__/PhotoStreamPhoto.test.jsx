// @flow

import { h } from 'preact';
import render from 'preact-render-to-string';
import PhotoStreamPhoto from '../';
import { photo } from '../../../testHelpers';

test('renders', () => {
  const string = render(<PhotoStreamPhoto startLoading={() => {}} id={photo.id} url={photo.small} alt={photo.title} />);
  expect(string).toMatchSnapshot();
});
