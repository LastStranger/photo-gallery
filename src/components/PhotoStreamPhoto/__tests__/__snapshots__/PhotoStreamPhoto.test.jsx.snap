// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<a class="stream-photo" href="/34266318003">
  <div aria-label="stray dog" role="img" class="stream-photo-img" style="background-image: url(https://farm5.staticflickr.com/4253/34266318003_64c6326099_m.jpg);"></div>
</a>
`;
